modules/javafx.graphics/src/main/java/com/sun/javafx/text/PrismTextLayout.java

```diff
@@ -944,11 +944,56 @@ private TextLine createLine(int start, int end, int startOffset) {
             leading = Math.max(leading, run.getLeading());
             length += run.getLength();
         }
+
+        /*
+         * Subtract the width of trailing spaces for the new TextLine so they're not taken into account
+         * for alignment calculations:
+         */
+
+        width -= computeTrailingSpaceWidth(startOffset, length, lineRuns);
+
         if (width > layoutWidth) layoutWidth = width;
         return new TextLine(startOffset, length, lineRuns,
                             width, ascent, descent, leading);
     }
 
+    private float computeTrailingSpaceWidth(int startOffset, int length, TextRun[] lineRuns) {
+        char[] chars = getText();
+        int trailingSpaces = 0;
+
+        for (int i = length + startOffset - 1; i >= startOffset; i--) {
+            if (chars[i] != ' ') {
```

Comment:
should `Character.isWhitespace()` be used instead?

Reply:
I'm not entirely sure, perhaps Phil Race @prrace could answer that?  There are loops that just check for 0x20, but also more complicated loops that use `Character.isWhitespace`.

Reply:
I think this logic needs to support all whitespace characters that might break.  Since there is no method in Character to indicate the breaking behavior, I wonder if we ought to either request one, or simply hardcode a list here.